### PR TITLE
Fix regression of 3411 - _lastUpdated gets clobbered during $reindex job

### PR DESCRIPTION
**Pull request by michaelabuckley**
This bug #3441  came back with the rewrite to the new job framework.

https://github.com/hapifhir/hapi-fhir/pull/3451

### CODE REVIEW

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/DateRangeUtil.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class DateRangeUtil {
+
+	/**
+	 * Narrow the DateRange to be within theStartInclusive, and theEndExclusive, if provided.
+	 * @param theDateRangeParam the initial range, null for unconstrained
+	 * @param theStartInclusive a lower bound to apply, or null for unchanged.
+	 * @param theEndExclusive an upper bound to apply, or null for unchanged.
+	 * @return a DateRange within the original range, and between theStartInclusive and theEnd
+	 */
+	@NotNull
```
Comment (tadgh) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/DateRangeUtil.java:18]: issue: I think we are meant to use javax.annotation.nonnull here.

File: hapi-fhir-base/src/test/java/ca/uhn/fhir/util/DateRangeUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		DateRangeParam result = DateRangeUtil.narrowDateRange(c.range, c.narrowStart, c.narrowEnd);
+
+		if (c.resultStart == null && c.resultEnd == null) {
+			assertThat(result, nullValue());
+		} else {
+			assertThat(result, notNullValue());
+			assertThat("range start", result.getLowerBound(), equalTo(c.resultStart));
+			assertThat("range end", result.getUpperBound(), equalTo(c.resultEnd));
+		}
+	}
```
Comment (tadgh) [inline hapi-fhir-base/src/test/java/ca/uhn/fhir/util/DateRangeUtilTest.java:110]: awesome test

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+type: fix
+issue: 3441, 3586
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml:3]: I _think_ comma-delimiting is safe here.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+jira: SMILE-3441
+title: "Regression of issue 3441. Reindexing jobs were not respected the passed in date range in SearchParams.
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml:5]:
```suggestion
title: "While converting the reindexing job to the new batch framework, a regression of [#3441](https://github.com/hapifhir/hapi-fhir/issues/3441) was introduced. Reindexing jobs were not respecting the passed in `_lastUpdated` parameter.
```

**Review by tadgh — APPROVED** [review]
Approved pending minor changes.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml (hunk 3 above)
Comment (michaelabuckley) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3586-lastIndex-reindex-fix.yaml:3]: I've dropped it be safe.

File: hapi-fhir-base/src/test/java/ca/uhn/fhir/util/DateRangeUtilTest.java (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-base/src/test/java/ca/uhn/fhir/util/DateRangeUtilTest.java:110]: Thx

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/DateRangeUtil.java (hunk 1 above)
Comment (michaelabuckley) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/util/DateRangeUtil.java:18]: I definitely did.  Fixed.

**Pull request merged**